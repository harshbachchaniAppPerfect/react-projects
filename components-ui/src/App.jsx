import * as React from "react";
import "./App.css";
import Pop from "./components/Pop.jsx";
import AccordionComp from "./components/AccordionComp.jsx";

function App() {
  return (
    <>
      <AccordionComp />
    </>
  );
}

export default App;
