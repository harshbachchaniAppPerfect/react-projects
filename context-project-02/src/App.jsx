import React, { useEffect, useState } from "react";
import ThemeBtn from "./components/ThemeButton.jsx";
import Card from "./components/Card.jsx";
import { ThemeProvider } from "./context/Theme.js";

function App() {
  const [themeMode, setThemeMode] = useState("light");

  const lightTheme = () => {
    setThemeMode("light");
  };
  const darkTheme = () => {
    setThemeMode("dark");
  };

  useEffect(() => {
    const mytag = document.querySelector("html").classList;
    mytag.remove("light", "dark");
    mytag.add(themeMode);
  }, [themeMode]);

  return (
    <ThemeProvider value={{ themeMode, lightTheme, darkTheme }}>
      <div className="flex flex-wrap min-h-screen items-center">
        <div className="w-full">
          <div className="w-full max-w-sm mx-auto flex justify-end mb-4">
            <ThemeBtn />
          </div>

          <div className="w-full max-w-sm mx-auto">
            <Card />
          </div>
        </div>
      </div>
    </ThemeProvider>
  );
}

export default App;
