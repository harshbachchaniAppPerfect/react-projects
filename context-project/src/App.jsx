import React from "react";
import UserContextProvider from "./context/UserContextProvider";
import Login from "./components/Login.jsx";
import Profile from "./components/Profile.jsx";
function App() {
  return (
    <div className="flex min-h-full flex-col justify-center px-6 py-12 lg:px-8">
      <UserContextProvider>
        <h1 className="mt-10 text-center text-4xl font-bold leading-9 tracking-tight text-gray-900">
          Context API Tutorial{" "}
        </h1>
        <Login />
        <Profile />
      </UserContextProvider>
    </div>
  );
}

export default App;
