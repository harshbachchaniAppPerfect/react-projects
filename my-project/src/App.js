import Card from "./components/Card.js";
import Home from "./components/Home.js";

function App() {
  let obj1 = {
    modelName: "MAC Air",
    chip: "M1",
  };
  let obj2 = {
    modelName: "MAC Pro",
    chip: "M2",
  };
  return (
    <>
      <h1>My First React App</h1>
      <Home />
      <br />
      <br />
      <br />
      <Card name="Mac" property={obj1} />
      <br />
      <Card name="Mac" property={obj2} />
    </>
  );
}

export default App;
