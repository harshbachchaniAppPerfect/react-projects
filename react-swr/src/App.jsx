import useSWR from "swr";
import { TodoProvider } from "./contexts/index.js";
import { TodoItem, TodoForm } from "./components/index.js";
import toast, { Toaster } from "react-hot-toast";
import { apigetAllTodos, todoUrlEndPoint as cacheKey } from "./api/todoApi.js";
import {
  addTodo,
  updateTodo,
  deleteTodo,
  toggleComplete,
} from "./utils/todoUtils.js";
import {
  addTodoOptions,
  updateTodoOptions,
  deleteTodoOptions,
  toggleTodoOptions,
} from "./api/todoAPISWR.js";

function App() {
  const getAllItems = async () => {
    try {
      const res = await apigetAllTodos();
      return res.data;
    } catch (error) {
      console.log(`Error is ${error}`);
      toast.error("Failed to Fetched items.", {
        duration: 1000,
      });
      throw error;
    }
  };
  const {
    isLoading,
    error,
    data: todos,
    mutate,
  } = useSWR(cacheKey, getAllItems, {
    onSuccess: (data) => data.sort((a, b) => b.id - a.id),
  });

  const mutateaddTodo = async (title) => {
    try {
      await mutate(
        addTodo(title),
        addTodoOptions({ id: 200, title, completed: false })
      );
      toast.success("Success! Added new item.", {
        duration: 1000,
        icon: "ðŸŽ‰",
      });
    } catch (error) {
      console.log(`Error is ${error}`);
      toast.error("Failed to add the new item.", {
        duration: 1000,
      });
    }
  };
  const mutateupdateTodo = async (id, title) => {
    try {
      await mutate(updateTodo(id, title), updateTodoOptions({ id, title }));
      toast.success("Success! Updated Item Successfully.", {
        duration: 1000,
        icon: "ðŸŽ‰",
      });
    } catch (error) {
      console.log(`Error is ${error}`);
      toast.error("Failed to update the item.", {
        duration: 1000,
      });
    }
  };
  const mutatedeleteTodo = async (id) => {
    try {
      await mutate(deleteTodo(id), deleteTodoOptions({ id }));
      toast.success("Success! Item deleted Successfully.", {
        duration: 1000,
        icon: "ðŸŽ‰",
      });
    } catch (error) {
      console.log(`Error is ${error}`);
      toast.error("Failed to delete the item.", {
        duration: 1000,
      });
    }
  };
  const mutatetoggleComplete = async (id) => {
    try {
      await mutate(toggleComplete(id), toggleTodoOptions({ id }));
      toast.success("Success! Item toggled Successfully.", {
        duration: 1000,
        icon: "ðŸŽ‰",
      });
    } catch (error) {
      console.log(`Error is ${error}`);
      toast.error("Failed to toggle item.", {
        duration: 1000,
      });
    }
  };

  if (error) {
    return <p>{error.message}</p>;
  }
  return (
    <TodoProvider
      value={{
        todos,
        mutateaddTodo,
        mutateupdateTodo,
        mutatedeleteTodo,
        mutatetoggleComplete,
      }}
    >
      <Toaster toastOptions={{ position: "top-center" }} />
      <div className="bg-[#172842] min-h-screen py-8">
        <div className="w-full max-w-2xl mx-auto shadow-md rounded-lg px-4 py-3 text-white">
          <h1 className="text-2xl font-bold text-center mb-8 mt-2">
            Manage Your Todos In My Web
          </h1>
          <div className="mb-4">
            <TodoForm />
          </div>
          <div className="flex flex-wrap gap-y-3">
            {isLoading ? (
              <p>Loading...</p>
            ) : (
              todos.map((todo) =>
                todo ? (
                  <div key={todo.id} className="w-full">
                    <TodoItem todo={todo} />
                  </div>
                ) : (
                  todos.pop()
                )
              )
            )}
          </div>
        </div>
      </div>
    </TodoProvider>
  );
}

export default App;
