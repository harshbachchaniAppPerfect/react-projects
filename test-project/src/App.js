import Item from "./components/items.js";
import MyList from "./components/List.js";
import MyTable from "./components/Mytable.js";

function App() {
  return (
    <>
      <MyList />
      <br />
      <br />
      <MyTable />
      <br />
      <br />
      <Item />
    </>
  );
}

export default App;
