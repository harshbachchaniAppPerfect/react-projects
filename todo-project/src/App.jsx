import React, { useReducer, useRef, useState } from "react";
import Todo from "./components/Todo.jsx";
import { initialTodos } from "./utils/todos.js";
import todosReducer from "./utils/todosReducer.js";
function App() {
  const [text, setText] = useState("");
  const [todos, dispatch] = useReducer(todosReducer, initialTodos);
  const myId = useRef(3);
  function handleAddTodo(text) {
    dispatch({
      type: "added",
      id: myId.current++,
      text: text,
    });
  }

  function handleChangeTodo(todo) {
    dispatch({
      type: "changed",
      todo: todo,
    });
  }

  function handleDeleteTodo(id) {
    dispatch({
      type: "deleted",
      id: id,
    });
  }
  return (
    <>
      <div className="container mx-auto my-10">
        <h1 className="text-center text-3xl font-semibold mb-4">ToDo List</h1>
        <div className="md:w-1/2 mx-auto">
          <div className="bg-white shadow-md rounded-lg p-6">
            <form>
              <div className="flex mb-4">
                <input
                  type="text"
                  className="w-full px-4 py-2 mr-2 rounded-lg
                             border-gray-300 focus:outline-none
                              focus:border-blue-500"
                  placeholder="Add new Task"
                  onChange={(e) => {
                    setText(e.target.value);
                  }}
                  value={text}
                />
                <button
                  onClick={(e) => {
                    e.preventDefault();
                    handleAddTodo(text);
                    setText("");
                  }}
                  className="bg-blue-500 hover:bg-blue-700 
                            text-white font-bold py-2 px-4 rounded"
                >
                  Add
                </button>
              </div>
            </form>
            <ul>
              {todos.map((todo) => {
                return (
                  <Todo
                    key={todo.id}
                    todo={todo}
                    handleChangeTodo={handleChangeTodo}
                    handleDeleteTodo={handleDeleteTodo}
                  />
                );
              })}
            </ul>
          </div>
        </div>
      </div>
    </>
  );
}

export default App;
