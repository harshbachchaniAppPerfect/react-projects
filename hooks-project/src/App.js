import "./App.css";
import Card from "./components/card.js";
import Form from "./components/form.js";
import MyForm from "./components/newform.js";
import ShapeEditor from "./components/shape.js";

function App() {
  return (
    <>
      <br />
      <br />
      <br />
      <h1 className="text-4xl text-center text-white">Password Generator</h1>
      <br />
      <br />
      <br />
      <Card />
      <br />
      <br />
      <br />
      <Form />
      <br />
      <br />
      <br />
      <ShapeEditor />
      <br />
      <br />
      <br />
      <MyForm />
    </>
  );
}

export default App;

/*
import { useState } from 'react';
import Chat from './Chat.js';
import ContactList from './ContactList.js';

export default function Messenger() {

  const [messages,setMessages]=useState(['','','']);
  const [to, setTo] = useState(contacts[0]);
  function addMessage(msg,contact){
    const newarray=[...messages];
    newarray[contact.id]=msg;
    setMessages(newarray)
  }
  return (
    <div>
      <ContactList
        contacts={contacts}
        selectedContact={to}
        onSelect={contact => setTo(contact)}
      />
      <Chat key={to.id} contact={to} onAdd={addMessage} message={messages[to.id]} />
    </div>
  )
}

const contacts = [
  { id: 0, name: 'Taylor', email: 'taylor@mail.com' },
  { id: 1, name: 'Alice', email: 'alice@mail.com' },
  { id: 2, name: 'Bob', email: 'bob@mail.com' }
];

import { useState } from 'react';

export default function Chat({ contact,message,onAdd }) {
  const [text, setText] = useState(message);
  return (
    <section className="chat">
      <textarea
        value={text}
        placeholder={'Chat to ' + contact.name}
        onChange={e => setText(e.target.value)}
      />
      <br />
      <button onClick={()=>onAdd(text,contact)}>Send to {contact.email}</button>
    </section>
  );
}


*/
