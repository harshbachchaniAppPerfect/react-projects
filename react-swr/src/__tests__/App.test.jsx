import { render, screen } from "@testing-library/react";
import { expect, test, vitest, describe } from "vitest";
import App from "../App.jsx";

test.skip.each([
  [1, 1, 2],
  [1, 2, 3],
  [2, 1, 3],
])("add(%i, %i) -> %i", (a, b, expected) => {
  expect(a + b).toBe(expected);
});

const person = {
  isActive: true,
  age: 32,
};

describe.skip("person", () => {
  test("person is defined", () => {
    expect(person).toBeDefined();
  });

  test("is active", () => {
    expect(person.isActive).toBeTruthy();
  });

  test("age limit", () => {
    expect(person.age).toBeLessThanOrEqual(32);
  });
});

test("Testing Snapshot", () => {
  const container = render(<App />);
  expect(container).toMatchSnapshot();
});

test("Test Heading", () => {
  render(<App />);
  // const text = screen.getAllByText("Manage Your Todos");
  const text = screen.getByText(/Manage YouR Todos/i); //skip case insensitive
  expect(text).toBeInTheDocument();
});
