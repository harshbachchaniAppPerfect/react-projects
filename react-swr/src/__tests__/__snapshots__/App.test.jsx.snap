// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Testing Snapshot 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        style="position: fixed; z-index: 9999; top: 16px; left: 16px; right: 16px; bottom: 16px; pointer-events: none;"
      />
      <div
        class="bg-[#172842] min-h-screen py-8"
      >
        <div
          class="w-full max-w-2xl mx-auto shadow-md rounded-lg px-4 py-3 text-white"
        >
          <h1
            class="text-2xl font-bold text-center mb-8 mt-2"
          >
            Manage Your Todos In My Web
          </h1>
          <div
            class="mb-4"
          >
            <form
              class="flex"
            >
              <h1
                class="text-blue-400 text-3xl"
              />
              <label
                for="AddTodo"
              >
                Add Todo
              </label>
              <input
                class="w-full border border-black/10 rounded-l-lg px-3 outline-none duration-150 bg-white/20 py-1.5"
                id="AddTodo"
                placeholder="Write Todo..."
                type="text"
                value=""
              />
              <button
                class="rounded-r-lg px-3 py-1 bg-green-600 text-white shrink-0"
                type="submit"
              >
                Add
              </button>
              <button
                class="rounded-r-lg px-3 py-1 bg-green-600 text-white shrink-0"
                type=""
              >
                Edit
              </button>
            </form>
          </div>
          <div
            class="flex flex-wrap gap-y-3"
          >
            <p>
              Loading...
            </p>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      style="position: fixed; z-index: 9999; top: 16px; left: 16px; right: 16px; bottom: 16px; pointer-events: none;"
    />
    <div
      class="bg-[#172842] min-h-screen py-8"
    >
      <div
        class="w-full max-w-2xl mx-auto shadow-md rounded-lg px-4 py-3 text-white"
      >
        <h1
          class="text-2xl font-bold text-center mb-8 mt-2"
        >
          Manage Your Todos In My Web
        </h1>
        <div
          class="mb-4"
        >
          <form
            class="flex"
          >
            <h1
              class="text-blue-400 text-3xl"
            />
            <label
              for="AddTodo"
            >
              Add Todo
            </label>
            <input
              class="w-full border border-black/10 rounded-l-lg px-3 outline-none duration-150 bg-white/20 py-1.5"
              id="AddTodo"
              placeholder="Write Todo..."
              type="text"
              value=""
            />
            <button
              class="rounded-r-lg px-3 py-1 bg-green-600 text-white shrink-0"
              type="submit"
            >
              Add
            </button>
            <button
              class="rounded-r-lg px-3 py-1 bg-green-600 text-white shrink-0"
              type=""
            >
              Edit
            </button>
          </form>
        </div>
        <div
          class="flex flex-wrap gap-y-3"
        >
          <p>
            Loading...
          </p>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
